modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java

```diff
@@ -18,429 +18,630 @@
-					devMod = childNodes.item(DevicePropertyIndex.DEVICE_MODEL.getValue()).
-							getChildNodes().item(SYNCML_ITEM_DATA_POSITION).getTextContent();
-					devLang = childNodes.item(DevicePropertyIndex.DEVICE_LANGUAGE.getValue()).
-							getChildNodes().item(SYNCML_ITEM_DATA_POSITION).getTextContent();
-
-					if (log.isDebugEnabled()) {
-						log.debug(
-								"OS Version:" + osVersion + ", IMSI: " + imsi + ", IMEI: " +
-										imei + ", DevID: " + devID + ", DevMan: " + devMan +
-										", DevMod: " + devMod + ", DevLang: " + devLang);
-					}
-
-					Device generatedDevice = generateDevice(DeviceManagementConstants.MobileDeviceTypes.
-									MOBILE_DEVICE_TYPE_WINDOWS, devID, osVersion, imsi, imei, devMan, devMod);
-					try {
-						SyncmlUtils.getDeviceManagementService().enrollDevice(generatedDevice);
-						return true;
-					} catch (DeviceManagementException e) {
-						String msg = "Exception while getting Device Management Service.";
-						log.error(msg, e);
-						throw new WindowsDeviceEnrolmentException(msg, e);
-					}
-				}
-			}
-		}
-
-		return false;
-	}
+    private static final int SYNCML_FIRST_MESSAGE = 1;
+    private static final int SYNCML_SECOND_MESSAGE = 2;
+    private static final int SESSIONID_FIRST = 1;
+    private static final int SESSIONID_SECOND = 2;
+    private static final int OSVERSION_POSITION = 0;
+    private static final int DEVICE_ID_POSITION = 0;
+    private static final int DEVICE_MODE_POSITION = 2;
+    private static final int DEVICE_MAN_POSITION = 1;
+    private static final int DEVICE_MOD_VER_POSITION = 3;
+    private static final int DEVICE_LANG_POSITION = 4;
+    private static final int IMSI_POSITION = 1;
+    private static final int IMEI_POSITION = 2;
+    private static final int VENDER_POSITION = 7;
+    private static final int MACADDRESS_POSITION = 8;
+    private static final int RESOLUTION_POSITION = 9;
+    private static final int DEVICE_NAME_POSITION = 10;
+    private static final String OS_VERSION = "OS_VERSION";
+    private static final String IMSI = "IMSI";
+    private static final String IMEI = "IMEI";
+    private static final String VENDOR = "VENDER";
+    private static final String MODEL = "DEVICE_MODEL";
+
+    List<? extends Operation> inProgressOperations;
+
+    private static Log log = LogFactory.getLog(SyncmlServiceImpl.class);
+
+    /**
+     * This method is used to generate and return Device object from the received information at
+     * the Syncml step.
+     *
+     * @param deviceID     - Unique device ID received from the Device
+     * @param osVersion    - Device OS version
+     * @param imsi         - Device IMSI
+     * @param imei         - Device IMEI
+     * @param manufacturer - Device Manufacturer name
+     * @param model        - Device Model
+     * @return - Generated device object
+     */
+    private Device generateDevice(String type, String deviceID, String osVersion, String imsi,
```

Comment:
already fixed it by creating bean for windows specific properties.
https://github.com/wso2/product-mdm/pull/154/